src/main/java/com/atlauncher/gui/tabs/accounts/AccountsTab.java

```diff
@@ -0,0 +1,553 @@
+                                "your password?"))
+                        .build())
+                    .setType(DialogManager.ERROR)
+                    .addOption(GetText.tr("Yes"), true)
+                    .addOption(GetText.tr("No"))
+                    .show();
+
+                if (ret != 0) {
+                    rememberField.setSelected(false);
+                }
+            }
+        });
+
+        gbc.gridx = 0;
+        gbc.gridy++;
+        gbc.gridwidth = 2;
+        Insets BOTTOM_INSETS = new Insets(10, 0, 0, 0);
+        gbc.insets = BOTTOM_INSETS;
+        gbc.anchor = GridBagConstraints.CENTER;
+        JPanel buttons = new JPanel();
+        buttons.setLayout(new FlowLayout());
+        leftButton = new JButton(GetText.tr("Add"));
+        leftButton.setName("leftButton");
+        leftButton.addActionListener(e -> login());
+        rightButton = new JButton(GetText.tr("Clear"));
+        rightButton.addActionListener(e -> {
+            if (accountsComboBox.getSelectedIndex() == 0) {
+                clearLogin();
+            } else {
+                int ret = DialogManager
+                    .yesNoDialog()
+                    .setTitle(GetText.tr("Delete"))
+                    .setContent(GetText.tr("Are you sure you want " +
+                        "to delete this account?"))
+                    .setType(DialogManager.WARNING).show();
+                if (ret == DialogManager.YES_OPTION) {
+                    viewModel.deleteAccount();
+                }
+            }
+        });
+        loginWithMicrosoftButton =
+            new JButton(GetText.tr("Login with Microsoft"));
+        loginWithMicrosoftButton.addActionListener(e -> {
+            // TODO This should be handled by some reaction via listener
+            int numberOfAccountsBefore = viewModel.accountCount();
+            new LoginWithMicrosoftDialog();
+
+            if (numberOfAccountsBefore !=  viewModel.accountCount()) {
+                viewModel.pushNewAccounts();
+                accountsComboBox.setSelectedItem(AccountManager.getSelectedAccount());
+            }
+        });
+        buttons.add(leftButton);
+        buttons.add(rightButton);
+        buttons.add(loginWithMicrosoftButton);
+        bottomPanel.add(buttons, gbc);
+
+        rightPanel.add(topPanel, BorderLayout.NORTH);
+        rightPanel.add(bottomPanel, BorderLayout.CENTER);
+
+        contextMenu = new JPopupMenu();
+
+        updateSkin = new JMenuItem(GetText.tr("Reload Skin"));
+        updateSkin.addActionListener(e -> {
+            viewModel.updateSkin();
+
+            // Have this done via listener
+            AbstractAccount account = viewModel.getSelectedAccount();
+            userSkin.setIcon(account.getMinecraftSkin());
+        });
+        contextMenu.add(updateSkin);
+
+        JMenuItem updateUsername =
+            new JMenuItem(GetText.tr("Update Username"));
+        updateUsername.addActionListener(e -> viewModel.updateUsername());
```

Comment:
Looks like this doesn't work. Just does nothing when clicked. Looks working in latest release

Reply:
I believe during development I thought this was something for mojang accounts. I was wrong, thus I had an instance check. I fixed that with [92372c1](https://github.com/ATLauncher/ATLauncher/pull/607/commits/92372c1862c5b751922a791fa0038a785ebc7f4c).